salt/states/pkg.py

```diff
@@ -1720,6 +1753,22 @@ def installed(
                         and x not in to_reinstall]
         failed = [x for x in failed if x in targets]
 
+        # Check for virtual packages in list of desired packages
+        if not sources:
+            try:
+                virtual_pkgs = []
+                for pkgname in [next(iter(x)) for x in pkgs] if pkgs else [name]:
+                    cver = new_pkgs.get(pkgname, [])
+                    if '1' in cver:
+                        virtual_pkgs.append(pkgname)
+                if virtual_pkgs:
+                    warnings.extend(_warn_virtual(virtual_pkgs))
+            except Exception:
+                # This is just some temporary code to warn the user about using
+                # virtual packages. Don't let an exception break the entire
+                # state.
+                pass
```

Comment:
Can we at leaast `log.debug` here this until the code goes away?

Reply:
Good idea, I've added debug logging here.